src/Tyrion/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -7,26 +7,24 @@ namespace Tyrion.Extensions
 {
     public static class ServiceCollectionExtensions
     {
-        public static IServiceCollection AddTyrion(this IServiceCollection services, Type assemblyTypeOrigin)
+        public static void AddTyrion(this IServiceCollection services, Type type)
         {
-            services.AddTransient<ITyrion, Tyrion>();
+            services.AddSingleton<ITyrion, Tyrion>();
```

Comment:
In this case you must always use scoped, because a singleton instance can broken you database context.